gui/src/2D/controls/inputTextArea.ts

```diff
@@ -0,0 +1,1282 @@
+import { Nullable } from "babylonjs/types";
+import { Observable } from "babylonjs/Misc/observable";
+import { Vector2 } from "babylonjs/Maths/math.vector";
+
+import { Control } from "./control";
+import { ValueAndUnit } from "../valueAndUnit";
+import { VirtualKeyboard } from "./virtualKeyboard";
+import { RegisterClass } from 'babylonjs/Misc/typeStore';
+import { Measure } from '../measure';
+import { InputText } from "./inputText";
+import { ICanvasRenderingContext, ITextMetrics } from "babylonjs/Engines/ICanvas";
+import { PointerInfoBase } from "babylonjs/Events/pointerEvents";
+import { IKeyboardEvent } from "babylonjs/Events/deviceInputEvents";
+
+/**
+ * Enum that determines the text-wrapping mode to use.
+ */
+export enum TextWrapping_ {
+    /**
+     * Clip the text when it's larger than Control.width; this is the default mode.
+     */
+    Clip = 0,
+
+    /**
+     * Wrap the text word-wise, i.e. try to add line-breaks at word boundary to fit within Control.width.
+     */
+    WordWrap = 1,
+
+    /**
+     * Ellipsize the text, i.e. shrink with trailing … when text is larger than Control.width.
+     */
+    Ellipsis,
+}
+
+/**
+ * Class used to create input text control
+ */
+export class InputTextArea extends InputText {
+
+    //private _textWrapping = TextWrapping_.Clip;
+    private _textWrapping = TextWrapping_.WordWrap;
+    //private _textHorizontalAlignment = Control.HORIZONTAL_ALIGNMENT_CENTER;
+    private _textHorizontalAlignment = Control.HORIZONTAL_ALIGNMENT_LEFT;
+    private _textVerticalAlignment = Control.VERTICAL_ALIGNMENT_TOP;
+
+    private _lines: any[];
+    private _oldlines: string[] = [];
+    private _clicked: boolean;
+    private _resizeToFit: boolean = false;
+    private _lineSpacing: ValueAndUnit = new ValueAndUnit(0);
+    private _outlineWidth: number = 0;
+    private _outlineColor: string = "white";
+
+    /**
+    * An event triggered after the text was broken up into lines
+    */
+    public onLinesReadyObservable = new Observable<InputTextArea>();
+
+    private lastClickedCoordinateY = 0;
+    private _selectedLineIndex = 0;
+    /** @hidden */
+    public _connectedVirtualKeyboard: Nullable<VirtualKeyboard>;
+    public measure: ITextMetrics;
```

Comment:
nit - naming. Measure might be confusing (as it is probably thought to be the measure of the control). maybe "textMeasurement"?

Reply:
Correct. It's changed to `textMetrics` to be complient with the type.